pkg/services/authn/authnimpl/sync/user_sync.go

```diff
@@ -305,14 +308,31 @@ func (s *UserSync) updateUserAttributes(ctx context.Context, usr *user.User, id
 		needsUpdate = true
 	}
 
+	// If the user is provisioned, we need to validate the authID
+	if usr.IsProvisioned {
+		s.log.Debug("User is provisioned", "id", id.Email)
+		needsConnectionCreation = false
+		authInfo, err := s.authInfoService.GetAuthInfo(ctx, &login.GetAuthInfoQuery{UserId: usr.ID, AuthModule: id.AuthenticatedBy})
+		if err != nil {
+			s.log.Error("Error getting auth info", "error", err)
+			return err
+		}
+
+		// Validate the authID matches the identity authId and the userUniqueID
+		if id.ExternalUID != authInfo.ExternalUID {
+			s.log.Error("provisioned authID mistmatches identity authID", "provisioned_authID", authInfo.ExternalUID, "identity_authID", id.ExternalUID)
```

Comment:
Please replace `authID` with `ExternalUID` and the message could be simpler like `ExternalUID mismatch`.

Reply:
Good eye! I rewrote this as per your suggestion